src/api/api_missions.nit

```diff
@@ -35,6 +36,32 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+
+		var post = req.body
+
+		var deserializer = new JsonDeserializer(post)
+		var submission = new Submission.from_deserializer(deserializer)
+		if not deserializer.errors.is_empty then
+			res.error 400
+			print "Error deserializing submitted mission: {post}"
+			return
+		end
+
+		var runner = config.engine_map[submission.engine]
+		var program = new Program(player, mission, submission.source)
+		runner.run(program)
+
+		program.update_status(config)
```

Comment:
You can also skip the whole `star update` in `update_status` if not success.

I think that it is an improvement to update the status on failures too, we might need to count the attempts, remember the last submission for the mission, etc.